packages/react-storage/src/components/StorageBrowser/actions/configs/types.ts

```diff
@@ -80,63 +84,93 @@ export interface TaskActionConfig<T extends TaskHandler> {
   /**
    * whether the action allows inflight cancellation
    */
-  isCancelable: IsCancelableTaskHandler<T>;
+  isCancelable: boolean;
 
   /**
    * action handler
    */
   handler: T;
 
   /**
-   * default title value displayed on action view
+   * default display name value displayed on action view
    */
-  title: string;
+  displayName: string;
+}
 
+export interface BatchTaskActionConfig<
+  T extends TaskHandler,
+  ComponentName extends Capitalize<string> = Capitalize<string>,
+> extends TaskActionConfig<T, ComponentName> {
   /**
    * sets action view subcomponent type and action processing behavior
    */
-  type: 'BATCH_ACTION' | 'SINGLE_ACTION';
+  type: 'BATCH_ACTION';
 }
 
-export interface ListActionConfig<T> extends ActionConfigTemplate<T> {}
+export interface SingleTaskActionConfig<
+  T extends TaskHandler,
+  ComponentName extends Capitalize<string> = Capitalize<string>,
+> extends TaskActionConfig<T, ComponentName> {
+  /**
+   * sets action view subcomponent type and action processing behavior
+   */
+  type: 'SINGLE_ACTION';
+}
 
-export interface UploadActionConfig extends TaskActionConfig<UploadHandler> {}
+export interface ListActionConfig<T, ComponentName>
+  extends ActionConfigTemplate<T> {
+  componentName: ComponentName;
```

Comment:
Can we just add `componentName` and the corresponding generic to `ActionConfigTemplate`?